packages/react-router/lib/components.tsx

```diff
@@ -222,37 +222,41 @@ export function Outlet(props: OutletProps): React.ReactElement | null {
   return useOutlet(props.context);
 }
 
-export interface PathRouteProps {
-  caseSensitive?: NonIndexRouteObject["caseSensitive"];
-  path?: NonIndexRouteObject["path"];
-  id?: NonIndexRouteObject["id"];
-  loader?: NonIndexRouteObject["loader"];
-  action?: NonIndexRouteObject["action"];
-  hasErrorBoundary?: NonIndexRouteObject["hasErrorBoundary"];
-  shouldRevalidate?: NonIndexRouteObject["shouldRevalidate"];
-  handle?: NonIndexRouteObject["handle"];
-  index?: false;
+export type PathRouteProps = Pick<
+  NonIndexRouteObject,
+  | "action"
+  | "caseSensitive"
+  | "handle"
+  | "hasErrorBoundary"
+  | "id"
+  | "loader"
+  | "path"
+  | "shouldRevalidate"
+> & {
+  index?: false; // this is also the same as in `NonIndexRouteObject`
   children?: React.ReactNode;
-  element?: React.ReactNode | null;
-  errorElement?: React.ReactNode | null;
-}
+  element?: React.ReactNode | null; // this is also the same as in `NonIndexRouteObject`
+  errorElement?: React.ReactNode | null; // this is also the same as in `NonIndexRouteObject`
+};
 
 export interface LayoutRouteProps extends PathRouteProps {}
 
-export interface IndexRouteProps {
-  caseSensitive?: IndexRouteObject["caseSensitive"];
-  path?: IndexRouteObject["path"];
-  id?: IndexRouteObject["id"];
-  loader?: IndexRouteObject["loader"];
-  action?: IndexRouteObject["action"];
-  hasErrorBoundary?: IndexRouteObject["hasErrorBoundary"];
-  shouldRevalidate?: IndexRouteObject["shouldRevalidate"];
-  handle?: IndexRouteObject["handle"];
-  index: true;
-  children?: undefined;
-  element?: React.ReactNode | null;
-  errorElement?: React.ReactNode | null;
-}
+export type IndexRouteProps = Pick<
+  IndexRouteObject,
+  | "action"
+  | "caseSensitive"
+  | "handle"
+  | "hasErrorBoundary"
+  | "id"
+  | "loader"
+  | "path"
+  | "shouldRevalidate"
+> & {
+  index: true; // this is also the same as in `IndexRouteObject`
+  children?: undefined; // this is also the same as in `IndexRouteObject`
+  element?: React.ReactNode | null; // this is also the same as in `IndexRouteObject`
+  errorElement?: React.ReactNode | null; // this is also the same as in `IndexRouteObject`
+};
```

Comment:
maybe its because I'm not a typescript wizard 😅 
I think the code on the left is might be more readable IMO than the newer code on the right?

Whats nice about the code left is:
- picking a type out of `IndexRouteObject` with JS-like syntax object
- jump to definition (cmd-click the string in square brackets) takes you immediately to the definition of the property; you can get a slightly similar experience with your code on the right by option-clicking `IndexRouteObject`, but it won't take you to the exact definition

Code on the right, is functionally the same, but I think you have to pause for a bit to parse whats happening with the `<Pick>` and `&` intersection; I know what both do, though

Minor stuff I don't have strong feelings about the change, just reading the code 

Are there other benefits to using `<Pick>`?

Reply:
For me the main benefit is not having to repeat  all these lines

The added benefit (although in a smaller way) is that it's immediately clear that you're picking certain properties from another type, completely like they are.
So no need to check if the child type is having the property mandatory/optional as well